topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsTopicStore.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic;
+
+import io.apicurio.registry.utils.kafka.ProducerActions;
+import io.vertx.core.Future;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiFunction;
+
+/**
+ * TopicStore based on Kafka Streams and
+ * Apicurio Registry's gRPC based Kafka Streams ReadOnlyKeyValueStore
+ */
+public class KafkaStreamsTopicStore implements TopicStore {
+    private static final Logger log = LoggerFactory.getLogger(KafkaStreamsTopicStore.class);
+
+    private final ReadOnlyKeyValueStore<String, Topic> topicStore;
+
+    private final String storeTopic;
+    private final ProducerActions<String, TopicCommand> producer;
+
+    private final BiFunction<String, String, CompletionStage<Integer>> resultService;
+
+    public KafkaStreamsTopicStore(
+            ReadOnlyKeyValueStore<String, Topic> topicStore,
+            String storeTopic,
+            ProducerActions<String, TopicCommand> producer,
+            BiFunction<String, String, CompletionStage<Integer>> resultService) {
+        this.topicStore = topicStore;
+        this.storeTopic = storeTopic;
+        this.producer = producer;
+        this.resultService = resultService;
+    }
+
+    public static Throwable toThrowable(Integer index) {
+        if (index == null) {
+            return null;
+        }
+        switch (index) {
+            case 1:
+                return new TopicStore.EntityExistsException();
+            case 2:
+                return new TopicStore.NoSuchEntityExistsException();
+            case 3:
+                return new TopicStore.InvalidStateException();
+        }
+        throw new IllegalStateException("Invalid index: " + index);
```

Comment:
shouldn't you move this into the switch as a default case?

Reply:
Moved ...